tgcrypto/ctr256.c

```diff
@@ -41,7 +41,8 @@ uint8_t *ctr256(const uint8_t in[], uint32_t length, const uint8_t key[32], uint
                 *state = 0;
 
             if (*state == 0) {
-                for (k = AES_BLOCK_SIZE - 1; k >= 0; --k)
+                k = AES_BLOCK_SIZE;
+                while(k--)
```

Comment:
I believe the `for` loop still reads better; we can simply adjust the range to `[0, AES_BLOCK_SIZE)` and correct the iv index to be `AES_BLOCK_SIZE - k - 1` instead.

Reply:
not sure if it reads better, since it requires more sugar for such a simple task, while loop fits just fine. Assembly-wise redundant subs each iteration seem not a good choice either

Reply:
Beside the unnoticeable difference in speed between the two variants, I still believe a for loop could fit better; concerns are also on variables' scopes, but since this is still C89 I guess we don't care much about that.
OK for the `while`, for now.

Reply:
If we really care both about scope AND C89 we'd have to make a new scope block both for `while` and `for` loops, but since k was in the function scope from the start I think it won't change much

Reply:
It's fine. Any variable had to be declared at the beginning of the scope anyway and having just one scope at this point, without explicit scope blocks, is ok.